src/lib/samples.rs

```diff
@@ -43,14 +44,14 @@ impl Sample {
     /// # Panics
     ///   - Panics if sample name is empty string.
     ///   - Panics if barcode is empty string.
-    ///   - Panics if barcode has bases other than A, C, G, or T.
+    ///   - Panics if barcode has bases other than A, C, G, T, or N/n/.
     #[must_use]
     pub fn new(ordinal: usize, name: String, barcode: String) -> Self {
```

Comment:
I suggest using [`bstr`](https://docs.rs/bstr/1.11.3/bstr/) rather than Strings - there's no reason to decode ASCII, especially when you just convert the barcode back to bytes.